pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -91,14 +93,15 @@ func ConvertService(svc coreV1.Service, domainSuffix string, clusterID cluster.I
 	}
 
 	istioService := &model.Service{
-		Hostname:        ServiceHostname(svc.Name, svc.Namespace, domainSuffix),
-		Ports:           ports,
-		Address:         addr,
-		ServiceAccounts: serviceaccounts,
-		MeshExternal:    meshExternal,
-		Resolution:      resolution,
-		CreationTime:    svc.CreationTimestamp.Time,
-		ClusterVIPs:     map[cluster.ID]string{clusterID: addr},
+		Hostname:              ServiceHostname(svc.Name, svc.Namespace, domainSuffix),
+		Ports:                 ports,
+		Address:               addr,
+		ServiceAccounts:       serviceaccounts,
+		MeshExternal:          meshExternal,
+		Resolution:            resolution,
+		CreationTime:          svc.CreationTimestamp.Time,
+		ClusterVIPs:           map[cluster.ID]string{clusterID: addr},
+		InternalTrafficPolicy: valueToStringGenerated(svc.Spec.InternalTrafficPolicy) == "Local",
```

Comment:
strconv.FormatBool

Reply:
> it is complicated

yes, here i deal it complicated

> You donot need this function to convert bool to string, it is complicated

Yes it can be simpler